sympy/geometry/point.py

```diff
@@ -111,6 +112,8 @@ def __new__(cls, *args, **kwargs):
         if len(coords) > dimension:
             if any(i for i in coords[dimension:]):
                 raise ValueError('too many nonzero dimensions')
+            if kwargs.get('warn', False):
+                raise ValueError('reducing dimensions')
```

Comment:
I think this should be a `warnings.warn` instead of an error.  Though the `warnings` module isn't often used in sympy, I think it should be :-)